### ISSUE TITLE
Better source code documentation - Questions about how it works

### ISSUE THREAD

**Issue by Vadorequest** [zeit/spr-landing#18]
I've read https://zeit.co/blog/serverless-pre-rendering and I'm interested by SPR, but I don't fully understand how it works.

AFAIU, everything is done there https://github.com/zeit/spr-landing/blob/master/pages/index.js#L177-L180

```js
  if (res) {
    res.setHeader("Cache-Control", "s-maxage=1, stale-while-revalidate");
    res.setHeader("X-version", etag);
  }
```

Few questions:
- What's the goal of `X-version` here? What happens if I don't define it?
- How does this cache implementation works? I don't understand how defining a 1s cache can change anything performance-wise.  
- Basically, considering I have a Next.js app which does SSR (fetching content from a headless CMS in `_app:getInitialProps`), what would happen if I just add those two lines in `_app/getInitialProps`?

**Comment by timneutkens** [comment]
You can read more about it here: https://zeit.co/blog/serverless-pre-rendering

**Comment by Vadorequest** [comment]
Yeah, I saw, this link is how I heard of this project (as I stated in my post), but I still don't understand the technicals at hand.

**Comment by Siilwyn** [comment]
Just dug into this, the cache control header just signals the CDN to cache the whole page as stated in the blog post. The `X-version` header & etag are not needed unless you want to refresh the page client-side when the displayed page is out of date.

**Comment by ijjk** [comment]
Hey folks! We have upgraded this repo to take advantage of incremental static site generation (iSSG). In short:

- There’s never a cold boot anymore when accessing the page
- The page gets incrementally re-built asynchronously
- Always fast & available

This is thanks to Next.js' new hooks for static site generation, [see the RFC here](https://github.com/zeit/next.js/issues/9524), where you can learn more about how this works.

I’m closing this issue for now as it’s no longer applicable. Feel free to ping me with any questions.
